### PR TITLE
X230 flash universal init

### PR DESCRIPTION
**Pull request by SebastianMcMillan**
This change makes the x230-flash.init file into a universal install.init, letting one reuse the install.init for other split SPI flash boards, such as the T430.

### CODE REVIEW

File: boards/x230-flash/x230-flash.config [hunk 1]
```diff
@@ -16,7 +16,7 @@ CONFIG_LINUX_USB=y
 CONFIG_LINUX_E1000E=y
 
 export CONFIG_USB_BOOT_DEV="/dev/sdb1"
-export CONFIG_BOOTSCRIPT=/bin/x230-flash.init
+export CONFIG_BOOTSCRIPT=/bin/install.init
```
Comment (merge) [inline boards/x230-flash/x230-flash.config]: "updating" and "x230-flash.config" seems to be wrong in the commit message.

File: initrd/bin/install.init [hunk 2]
```diff
@@ -20,7 +20,7 @@ echo ''
 echo 'To install from flash drive:'
 echo ''
 echo '  mount -o ro /dev/sdb1 /media'
-echo '  flashrom-x230.sh /media/x230.rom'
+echo '  flash.sh /media/<rom-file-name>'
 echo ''
```
Comment (merge) [inline initrd/bin/install.init]: has already been fixed...

File: boards/x230-flash/x230-flash.config (hunk 1 above)
Comment (merge) [reply boards/x230-flash/x230-flash.config]: oh sorry, I was reading it wrong. the commit message is fine :)

**Comment by merge** [comment]
I would prefer a new <laptop-model>-flash.init script. And if we then see that it really can be exactly the same for different models, we can rename.

Adding support for a new board itself would be nice not to touch others. Is it just me?

File: initrd/bin/install.init (hunk 2 above)
Comment (SebastianMcMillan) [reply initrd/bin/install.init]: it wasn't back when those changes were actually made.

**Comment by SebastianMcMillan** [comment]
Seeing how the x230-flash.init doesn't do anything specific to the x230, I figure it would be easier to make a universal install.init, not only reducing the amount of clutter we have in the initrd, but also is easier to work with.

**Comment by tlaurion** [comment]
@merge flash specifics are defined under flash.sh anyway. Each board addition should be added to it anyway. I like the idea to generalize!

File: initrd/bin/install.init (hunk 2 above)
Comment (tlaurion) [reply initrd/bin/install.init]: > has already been fixed...

@merge, where?

**Comment by tlaurion** [comment]
> I would prefer a new -flash.init script. And if we then see that it really can be exactly the same for different models, we can rename.
> 
> Adding support for a new board itself would be nice not to touch others. Is it just me?

@merge: I would love to better understand your stance following [past comment](https://github.com/osresearch/heads/pull/579#issuecomment-500083390)

File: initrd/bin/install.init (hunk 2 above)
Comment (merge) [reply initrd/bin/install.init]: in https://github.com/osresearch/heads/commit/00bf4740e3c08feb1870f4c7b4dd7823681bf22c

**Comment by merge** [comment]
> > I would prefer a new -flash.init script. And if we then see that it really can be exactly the same for different models, we can rename.
> > Adding support for a new board itself would be nice not to touch others. Is it just me?
> 
> @merge: I would love to better understand your stance following [past comment](https://github.com/osresearch/heads/pull/579#issuecomment-500083390)

I think x230-flash.init can indeed be renamed to something more generic, if we add another "xxxx-flash" board.

**Comment by tlaurion** [comment]
> > > I would prefer a new -flash.init script. And if we then see that it really can be exactly the same for different models, we can rename.
> > > Adding support for a new board itself would be nice not to touch others. Is it just me?
> > 
> > 
> > @merge: I would love to better understand your stance following [past comment](https://github.com/osresearch/heads/pull/579#issuecomment-500083390)
> 
> I think x230-flash.init can indeed be renamed to something more generic, if we add another "xxxx-flash" board

@merge: and my point is that the boards specific flashroms options currently need to be defined under [flash.sh](https://github.com/osresearch/heads/blob/master/initrd/bin/flash.sh#L14), the proper flashrom call being defined per $CONFIG_BOARD variable, put dynamically under /etc/config, sourced by init and all other scripts under Heads.

The only current exception in boards being if the board in question can flash another SPI (eg: iKVM) from within, like in the case of the [kgpe-d16](https://github.com/osresearch/heads/blob/master/initrd/bin/flashrom-kgpe-d16-openbmc.sh)

**Comment by tlaurion** [comment]
@merge @SebastianMcMillan update?

**Comment by tlaurion** [comment]
@SebastianMcMillan please rebase.
@MrChromebox : I think this should be merged, personally.

**Comment by MrChromebox** [comment]
@tlaurion needs a rebase obv since the one commit isn't needed anymore. I think 'install.init' isn't the best name, but I can live with it

**Comment by SebastianMcMillan** [comment]
If the plan is to drop the 2-part flash in favor of flashing both chips externally, then I don't think this PR should be merged.

**Comment by tlaurion** [comment]
@SebastianMcMillan gbe and me blobs are still nit.resolved problems.

File: boards/x230-flash/x230-flash.config (hunk 1 above)
Comment (tlaurion) [reply boards/x230-flash/x230-flash.config:20]: @SebastianMcMillan This part would still be nice.

**Comment by SebastianMcMillan** [comment]
Now that I've got an X230 again, I'll have to take a look at it sometime
soon.

**Pull request closed without merging**